hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -108,31 +108,38 @@ checkAbort ctx@ScriptContext{scriptContextTxInfo = txInfo} headCurrencySymbol pa
       Nothing -> 0
       Just tokenMap -> negate $ sum tokenMap
 
-  -- FIXME: This is prone to identical commits only being reimbursed once.
   mustReimburseCommittedUTxO =
-    traceIfFalse "committed UTxO not reimbursed" $
-      all (`elem` serialisedOutputs) committedUTxOs
+    traceIfFalse "hash of committed UTxo do not match hash of outputs" $
+      hashOfCommittedUTxO == hashOfOutputs
 
-  serialisedOutputs = Builtins.serialiseData . toBuiltinData <$> txInfoOutputs txInfo
+  hashOfOutputs =
+    -- NOTE: It is enough to just _take_ the same number of outputs that
+    -- correspond to the number of commit inputs to make sure everything is
+    -- reimbursed because we assume the outputs are correctly sorted with
+    -- reimbursed commits coming first
+    hashTxOuts $ take (length commited) (txInfoOutputs txInfo)
```

Comment:
Using `length committed` (btw a typo) is actually a good point here and we should redefine it in the spec (is a redeemer there right now)